src/main.rs

```diff
@@ -148,47 +149,68 @@ async fn main() -> miette::Result<()> {
                 PackageType::Impl
             };
 
-            command::init(kind, package)
+            command::init(kind, package.to_owned())
                 .await
-                .wrap_err(miette!("init command failed"))
+                .wrap_err(miette!(
+                    "failed to initialize {}{kind} package",
+                    package.map(|p| format!("`{p}` as ")).unwrap_or_default()
+                ))
         }
         Command::Add {
             registry,
             dependency,
-        } => command::add(registry, &dependency)
+        } => command::add(registry.to_owned(), &dependency)
             .await
-            .wrap_err(miette!("add command failed")),
-        Command::Remove { package } => command::remove(package)
-            .await
-            .wrap_err(miette!("remove command failed")),
+            .wrap_err(miette!(
+                "failed to add `{dependency}` from `{registry}` to `{MANIFEST_FILE}`"
```

Comment:
One thing I don't like is that if you pass a nonsense value for `dependency` it will still say "nonsense from {registry}", even though clearly it's not from there. The lower level messages may clarify but it's still a bit misleading.